pkg/kind/kong_proxy.go

```diff
@@ -109,50 +100,67 @@ func (c *kongProxyCluster) startDeploymentInformer(ctx context.Context) (deploym
 	return
 }
 
-func (c *kongProxyCluster) startServiceInformer(ctx context.Context, nsn types.NamespacedName, ready chan ProxyReadinessEvent) {
-	factory := kubeinformers.NewSharedInformerFactory(c.client, proxyInformerResyncPeriod)
-	informer := factory.Core().V1().Services().Informer()
-	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		// this UpdateFunc is responsible for closing the ready channel once the proxy URL is ready (or if an error occurs)
-		UpdateFunc: func(old1, new1 interface{}) {
-			newService, ok := new1.(*corev1.Service)
-			if !ok {
-				err := fmt.Errorf("somehow got unexpected type instead of corev1.Service: %T", new1)
+// startServiceInformer exposes the provided deployment as LoadBalancer type and reports when the service has been provisioned a load balancer IP/Host
+func (c *kongProxyCluster) startServiceInformer(ctx context.Context, d *appsv1.Deployment, ready chan ProxyReadinessEvent, timeout time.Time) {
```

Comment:
I have several comments about how this function deals with contexts and timeouts:
- It appears that idiomatically in Go ["timeouts" are of type `time.Duration`](https://pkg.go.dev/context#WithDeadline) and ["deadlines" are of type `time.Time`](https://pkg.go.dev/context#WithDeadline). Here a `timeout` is a `time.Time` which left me confused.
- This function accepts a `ctx` (and passes it on) unrelated to the deadline. The deadline does not affect blocked operations accepting a context, but it could, virtually for free.
- There's a pattern for ticks with a deadline:

    ```go
    for timeNow := time.Tick(serviceInformerTickTime) {
      // option 1
      if timeNow.After(deadline) {
        break
      }
      // option 2
      select {
      case <-ctx.Done():
        break
      }
    }
    ```

My recommendation would be to drop the deadline altogether, and instead use `ctx`'s capabilities to implement timeout in this function.

Reply:
My only concern has to do with one big timeout for an entire test (via the context) vs. timeouts for specific portions which can be more visible and tunable... :thinking: 

But I think for now I'll take your recommendation and we'll test with that for a while :+1:

Reply:
After looking things over, I want to hold on making changes right now since this is blocking another PR and is otherwise functional, I created a follow up item to clean this up before our first release version: https://github.com/Kong/kubernetes-testing-framework/issues/14